src/ShaderGen/ShaderFunctionAndMethodDeclarationSyntax.cs

```diff
@@ -27,5 +27,13 @@ public bool Equals(ShaderFunctionAndMethodDeclarationSyntax other)
             return Function.DeclaringType == other.Function.DeclaringType
                 && Function.Name == other.Function.Name;
         }
+
+        public override int GetHashCode()
+        {
+            var hashCode = 1204124163;
+            hashCode = hashCode * -1521134295 + Function.DeclaringType.GetHashCode();
```

Comment:
Any particular reference for this formula? It's not super important, just curious.